forem/forem
# Allow admins to delete an organization

## What type of PR is this? (check all applicable)


## Description

Being able to delete organizations as a super_admin. Currently, a worker exists that a user of an organization will use to delete the organization it belongs to. As a result there is some logic that exists in the worker that relates to updating and notifying the user about the deletion. 

I also changed the interface to look like the member manager page. It was a quick change and I had Anuj's blessings 😅 

Note: if an organization has credits available they cannot be deleted. 

## Related Tickets & Documents


- Related Issue #https://github.com/forem/forem/issues/16041
- Closes #https://github.com/forem/forem/issues/16041

## QA Instructions, Screenshots, Recordings

Demo can be found [here](https://www.loom.com/share/93793a0255624abc87aa0e31ed402ca6?sid=4cb462d5-7f32-4bd4-8eaa-104f28150df5): 

- Go to admin/content_manager/organizations
- Click on one of the organizations 
- Click on the three dot menu on the right hand side 
- Click on delete organization 
- If you are a super admin you should see a message that indicates that the we've schedule a job to delete it. 
- If you are not a super admin you should see an error that you need super admin permissions
- If the organization has credit associated then you should see an error that the organization has credits associated with it.

Some screenshots: 

<img width="1512" alt="Screenshot 2023-07-11 at 15 00 33" src="https://github.com/forem/forem/assets/2786819/a10004cc-c4be-43d1-b2a4-0d6b38a48f65">

If you are not a super admin you should see an error that you need super admin permissions and If the organization has credit associated then you should see an error that the organization has credits associated with it:

<img width="1512" alt="Screenshot 2023-07-11 at 15 01 03" src="https://github.com/forem/forem/assets/2786819/f8a58a47-ea81-4a10-8e1f-ec9225c543ea">

Deleting an organization:
<img width="1512" alt="Screenshot 2023-07-11 at 15 02 29" src="https://github.com/forem/forem/assets/2786819/e9301e56-3b9f-4d8d-bacd-8ca6a89f061b">

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

      have not been included_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/v6aOjy0Qo1fIA/giphy.gif)
